### DISCUSSION TITLE
Using MySQL functions inside DataObjects as parts of the query

### DISCUSSION THREAD

**Discussion by AleksandarV993** [pimcore/pimcore#14611] [Q&A / Support] [answered]
Hello,

I was wondering if there's a way to use default MySQL functions inside `DataObjects`? For example, how would I query for `MIN(price)` and/or `MAX(price)` while using the default methods which the `DataObject` offers? I'm aware that I can set conditions like:
```
$do = new Pimcore\Model\DataObject\MyClass;
$do->setCondition("price = 100");
```
But that only covers WHERE, WHERE AND and WHERE OR. How would I use `DATE`, `MIN`, `MAX` and other functions while using the default `DataObject` model? 

If this is not possible, would it be possible to append RAW SQL fields to the `DataObject` model in a matter like `->select('MIN(price)')` for example?

Thanks in advance,
Alex

**Comment by twin-elements** [comment, marked as answer]
Hi. 
You can use listing and query builder:
```
$queryBuilder = (new Pimcore\Model\DataObject\MyClass\Listing())->getQueryBuilder();
$queryBuilder->addSelect('MIN(price)');
```

Example
```
$minMaxPrices = (new Pimcore\Model\DataObject\Product\Listing())->getQueryBuilder()
                ->addSelect('MIN(price) as min')
                ->addSelect('MAX(price) as max')
                ->setMaxResults(1)
                ->execute();
```
